src/internal/analytics-metadata/__tests__/labels-utils.test.tsx

```diff
@@ -225,6 +225,43 @@ describe('processLabel', () => {
     expect(processLabel(target, { selector: '.outer-class', root: 'body' })).toEqual('label outside of the component');
   });
 
+  test('respects the rootClassName property', () => {
+    const { container } = render(
+      <div className="root-class">
+        <div className="label-class">outer label</div>
+        <div id="target">
+          <div className="label-class">inner label</div>
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+    expect(processLabel(target, { selector: '.label-class', rootClassName: 'root-class' })).toEqual('outer label');
+  });
+  test('rootClassName prevails over root property', () => {
+    const { container } = render(
+      <>
+        <div className="root-class">
+          <div className="label-class">root class label</div>
+          <div {...getAnalyticsMetadataAttribute({ component: { name: 'ComponentName' } })}>
+            <div className="label-class">component label</div>
+            <div id="target">
+              <div className="label-class">inner label</div>
+            </div>
+          </div>
+        </div>
+        <div className="outer-class">label outside of the component</div>
+      </>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+    expect(processLabel(target, { selector: '.label-class', root: 'self', rootClassName: 'root-class' })).toEqual(
+      'root class label'
+    );
+    expect(processLabel(target, { selector: '.label-class', root: 'component', rootClassName: 'root-class' })).toEqual(
+      'root class label'
+    );
+    expect(processLabel(target, { selector: '.outer-class', root: 'body', rootClassName: 'root-class' })).toEqual('');
```

Comment:
question: what root: 'body' is for?

Reply:
for the document body. I had to introduce it to identify group labels in dropdown-based components when expandToViewport=true (https://github.com/cloudscape-design/component-toolkit/pull/94)

Reply:
Why is it expected for this one to result to an empty string, shouldn't the rootClassName prevail?

Reply:
It does prevail. The method returns an empty string because there is no element with class `outer-label` inside the `root-class` element.